pyro/contrib/tracking/assignment.py

```diff
@@ -0,0 +1,152 @@
+from __future__ import absolute_import, division, print_function
+
+import itertools
+import math
+import numbers
+import warnings
+
+import torch
+
+import pyro.distributions as dist
+
+
+def _warn_if_nan(tensor, name):
```

Comment:
I've wanted to put this in a `pyro.debug` module so it can be exposed as a tool for users but we can do that before release

Reply:
agreed. we're also using it in [pyro.ops.newton](https://github.com/uber/pyro/blob/dev/pyro/ops/newton.py#L9)